packages/module-ui--selection-mode/src/lib/selection-mode-manager.spec.ts

```diff
@@ -0,0 +1,205 @@
+import { SelectionModeManager } from './selection-mode-manager.js';
+import { SelectionModeOptions } from './types.js';
+import { ISidebarTabSwitchController } from '@wme-enhanced-sdk/module-ui--sidebar-manager';
+
+// Mock SDK
+const createMockSdk = () => ({
+  Editing: {
+    setSelection: vi.fn(),
+    getSelection: vi.fn(),
+    clearSelection: vi.fn(),
```

Comment:
We don't have clearSelection in the SDK, but setSelection with null. And it'd cause a tab switch, so maybe that's good to test as well, to test the sidebar tab won't be changed and that it'd be intercepted.